### PR TITLE
Shared CSS, CSS Variables, and Grid Setup

### PR DESCRIPTION
**Pull request by nabeelvalley**
### Description of Changes:

- Correctly expose components via component index using TypeScript
- Add and document shared CSS for:
    - Typography
    - Global Variables
    - Grid
- Move `pull_request_template` to `.github` directory

### Checklist
- [x] Does it compile?
- [x] Are all tests passing?
- [x] Is all relevant documentation added?

### CODE REVIEW

File: src/pages/_app.tsx [hunk 1]
```diff
@@ -4,6 +4,7 @@ import NextApp, { AppContext } from 'next/app';
 import React from 'react';
 import { Provider } from 'react-redux';
 import makeStore from '../store';
+import '../styles/index.css';
```
Comment (DeanMW) [inline src/pages/_app.tsx:7]: Nothing to change here for now just a note that this should be added to the aliases in the future.

File: src/styles/styles.mdx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+```
+
+Additionally the variable naming should be set using a combination of the variable type and descriptor for the value
+
+| Type  | Descriptor | Variable Name    |
+| ----- | ---------- | ---------------- |
+| Font  | Main       | `--font-main`    |
+| Color | Header     | `--color-header` |
+
+> Note that the naming of variables should be done using the American spelling for `color`
```
Comment (DeanMW) [inline src/styles/styles.mdx:55]: hahaha 😛 I guess this is it now

File: src/styles/variables.css [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+  /* Spacing */
+  --spacing-main-horizontal: 10%;
+  --spacing-base: 4px;
+  --spacing-xs: 4px;
+  --spacing-s: 8px;
+  --spacing-m: 16px;
+  --spacing-l: 24px;
+  --spacing-xl: 36px;
+}
```
Comment (DeanMW) [inline src/styles/variables.css:19]: nice this is exactly what i was hoping for 😬

File: src/pages/_app.tsx (hunk 1 above)
Comment (nabeelvalley) [reply src/pages/_app.tsx:7]: @DeanMW we can potentially do that, however, this file should only ever be imported in the `_app.tsx` file, so we'll never actually import the global styles anywhere else

**Comment by nabeelvalley** [comment]
Thanks @biltongza , @DeanMW. Merging this in now

**Pull request merged**